src/BoundaryConditions/boundary_condition.jl

```diff
@@ -144,9 +144,13 @@ validate_boundary_condition_architecture(bc::BoundaryCondition, arch, side) =
 validate_boundary_condition_architecture(condition, arch, bc, side) = nothing
 validate_boundary_condition_architecture(::Array, ::CPU, bc, side) = nothing
 validate_boundary_condition_architecture(::CuArray, ::GPU, bc, side) = nothing
+validate_boundary_condition_architecture(::MtlArray, ::MetalBackend, bc, side) = nothing
 
-validate_boundary_condition_architecture(::CuArray, ::CPU, bc, side) =
+validate_boundary_condition_architecture(::CuArray, ::Union{CPU, MetalBackend}, bc, side) =
```

Comment:
```suggestion
validate_boundary_condition_architecture(::CuArray, arch, bc, side) =
```

would it be better to make the fallback error and then whitelist `CuArray` + `GPU`? Then we don't have to put in all the cases that don't work manually.

Reply:
I wasn't sure if there was a reason the fallback was different?

Reply:
Previously there were only two cases, so there's no distinction between a "whitelist" or "blacklist". But with 3 cases (this PR), that distinction emerges